### PR TITLE
Only consider one metric when averaging checkpoints

### PR DESCRIPTION
**Pull request by KellenSunderland**
Metrics files have added additional columns since this code was written.  It is now required to select the column that you wish to use by name.  

Below is an example of a previously present line:

```1       bleu-val=0.001626       perplexity-train=454.856855     perplexity-val=266.975737       time-elapsed=1014.405378```

With the previous code this would have added two values (blue-val + perplexity-val) which would have made selecting a true top-n result based on one of them difficult.

### CODE REVIEW

File: sockeye/average.py [hunk 1]
```diff
@@ -146,7 +149,7 @@ def read_metrics_points(path: str) -> Iterable[Tuple[float, int]]:
                 key_value = field.split("=")
```
Comment (tdomhan) [inline sockeye/average.py]: would it make sense to fail if we don't find the metric anywhere?

Comment (KellenSunderland) [inline sockeye/average.py]: Yes, failing with a descriptive error message would probably be a good idea.

File: sockeye/average.py [hunk 2]
```diff
@@ -166,7 +169,12 @@ def main():
         help="either a single model directory (automatic checkpoint selection) "
              "or multiple .params files (manual checkpoint selection)")
     params.add_argument(
-        "--max", action="store_true", help="maximize metric (default: min)")
+        "--max", action="store_true", help="Maximize metric.")
+    params.add_argument(
+        "--metric",
+        help="Name of the metric to choose n-best checkpoints from. (default: {})".format(C.PERPLEXITY),
```
Comment (fhieber) [inline sockeye/average.py]: in the help message you can write : "Default: %(default)s" to automatically take over what is set as default in kwargs (see arguments.py)

File: sockeye/average.py [hunk 3]
```diff
@@ -166,7 +169,12 @@ def main():
...
+        default=C.PERPLEXITY,
+        choices=[C.PERPLEXITY, C.BLEU])
```
Comment (fhieber) [inline sockeye/average.py]: I think we can also add accuarcy here (its not tracked by default anymore but is generally supported)

**Pull request merged**